sway-core/src/type_system/ast_elements/type_parameter.rs

```diff
@@ -1252,3 +1265,23 @@ impl IsConcrete for ConstGenericParameter {
         todo!("Will be implemented by https://github.com/FuelLabs/sway/issues/6860")
     }
 }
+
+impl PartialEq for ConstGenericParameter {
+    fn eq(&self, other: &Self) -> bool {
+        self.name.as_str() == other.name.as_str()
```

Comment:
Is `PartialEq` actually used anywhere? I mean, `PartialEqWithEngines` is what usually "matters". I am asking because here we consider two `ConstGenerisParameter`s as equal if their names match. But, e.g., different types ids and other fields. Of course, two different type ids (the `ty` field) can be same with engines. I am just wondering if this simple equality could cause clashes anywhere?